tests/test_coredump.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+import subprocess
+import os
+import socket
+import json
+from shlex import quote
+from pathlib import Path
+from typing import List, Any
+from dataclasses import dataclass
+
+import conftest
+
+
+@dataclass
+class Vm:
+    kernel: Path
+    squashfs: Path
+    initial_ramdisk: Path
+    kernel_params: List[str]
+
+
+def rootfs_image(image: Path):
+    result = subprocess.run(
+        ["nix-build", str(image), "-A", "json"], text=True, stdout=subprocess.PIPE, check=True
+    )
+    with open(result.stdout.strip("\n")) as f:
+        data = json.load(f)
+        return Vm(
+            kernel=Path(data["kernel"]),
+            squashfs=Path(data["squashfs"]),
+            initial_ramdisk=Path(data["initialRamdisk"]),
+            kernel_params=data["kernelParams"],
+        )
+
+def is_open(ip,port):
+   s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+   try:
+      s.connect((ip, int(port)))
+      s.shutdown(2)
+      return True
+   except:
+      return False
+
+class Qemu:
+    def __init__(self, vm: Vm):
+        self.vm = vm
+
+    def attach(self):
+        """
+        Attach to qemu session via tmux. This is useful for debugging
+        """
+        subprocess.run(["tmux", "-L", self.tmux_session, "attach"])
```

Comment:
@pogobanane if you use qemu.attach() this give you access to the VM. This makes debugging tests easier.